static/src/javascripts-legacy/projects/common/modules/check-dispatcher.js

```diff
@@ -4,28 +4,59 @@ define([
     'common/modules/check-mediator',
     'common/modules/email/run-checks',
     'common/modules/email/email-article',
-    'common/modules/experiments/ab-test-clash'
+    'common/modules/experiments/ab-test-clash',
+    'commercial/modules/dfp/track-ad-render',
 ], function(
     find,
     config,
     checkMediator,
     emailRunChecks,
     emailArticle,
-    clash
+    clash,
+    trackAdRender
 ) {
 
     var checksToDispatch = {
-        isUserInAClashingAbTest: function() {
-            return clash.userIsInAClashingAbTest();
+        isUserInContributionsAbTest: function() {
+            return clash.userIsInAClashingAbTest(clash.contributionsTests);
         },
-        emailCanRun: function() {
+        isUserNotInContributionsAbTest: function() {
+            return checkMediator.waitForCheck('isUserInContributionsAbTest').then(function (userInContributionsAbTest) {
+                return !userInContributionsAbTest;
+            });
+        },
+        isUserInEmailAbTest: function() {
+            return clash.userIsInAClashingAbTest(clash.emailTests);
+        },
+        emailCanRunPreCheck: function() {
             return emailRunChecks.allEmailCanRun();
         },
         listCanRun: function() {
             return !!find(emailArticle.getListConfigs(), emailRunChecks.listCanRun);
         },
         emailInArticleOutbrainEnabled: function() {
             return config.switches.emailInArticleOutbrain;
+        },
+        hasHighPriorityAdLoaded: function() {
+            return trackAdRender('dfp-ad--merchandising-high').then(function(highPriorityAdLoaded) {
+                return highPriorityAdLoaded;
+            });
+        },
+        hasLowPriorityAdLoaded: function() {
+            return checkMediator.waitForCheck('hasHighPriorityAdLoaded').then(function (highPriorityAdLoaded) {
+                if (highPriorityAdLoaded) {
+                    return trackAdRender('dfp-ad--merchandising').then(function(lowPriorityAdLoaded) {
```

Comment:
Yes the return value of this function is used to resolve a promise in check-mediator, you can resolve that promise with a boolean or another promise, so it's OK.